comms.py

```diff
@@ -0,0 +1,81 @@
+from mcu import *
+
+class Communications:
+    
+    PWM_MIN = 1000
+    PWM_MID = 1500
+    PWM_MAX = 2000
+    
+    MOTOR_CLAW = 0 
+    MOTOR_FRONT = 1
+    MOTOR_RIGHT = 2
+    MOTOR_LEFT = 3 
+    MOTOR_BACK = 4
+    
+    INITIAL_PERCENT = 100
+    
+    def __init__(self, mcuVAR: MCUInterface, MULTIPLIER_PERCENT: int):
+        self.mcuVAR = mcuVAR
+        self.MULTIPLIER_PERCENT = MULTIPLIER_PERCENT
+
+    def read_send(key_pressed):
+        
+        self.mcuVAR.cmd_setMotorCalibration(MOTOR_CLAW,1000)
+        self.mcuVAR.cmd_setMotorCalibration(MOTOR_FRONT, 1000)
+        self.mcuVAR.cmd_setMotorCalibration(MOTOR_LEFT, 1000)
+        self.mcuVAR.cmd_setMotorCalibration(MOTOR_RIGHT, 1000)
+        self.mcuVAR.cmd_setMotorCalibration(MOTOR_BACK, 1000)
+
+        self.mcuVAR.cmd_setMotorMicroseconds(MOTOR_CLAW, PWM_MID)
+        self.mcuVAR.cmd_setMotorMicroseconds(MOTOR_FRONT, PWM_MID)
+        self.mcuVAR.cmd_setMotorMicroseconds(MOTOR_LEFT, PWM_MID)
+        self.mcuVAR.cmd_setMotorCalibration(MOTOR_RIGHT, PWM_MID)
+        self.mcuVAR.cmd_setMotorCalibration(MOTOR_BACK, PWM_MID)
+
+        
+        dict_motors = {
+            "e": [MOTOR_FRONT, MOTOR_BACK, OPPOSITE],
+            "q": [MOTOR_BACK, MOTOR_FRONT, OPPOSITE],
+            "d": [MOTOR_RIGHT, MOTOR_LEFT, SAME],
+            "a": [MOTOR_RIGHT, MOTOR_LEFT, NEG_SAME]
+            "w": [MOTOR_LEFT, MOTOR_RIGHT, OPPOSITE],
+            "s": [MOTOR_RIGHT, MOTOR_LEFT, OPPOSITE] 
+        }
+        
+        spacebar_count = 0
+
+        if key_pressed[0] == "s" and len(key_pressed) == 2:
+            for motor in range(0, len(dict_motors[key_pressed[1]])-1):
+                self.mcuVAR.cmd_setMotorCalibrated(dict_motors[key_pressed[1]][motor],0)
+        elif key_pressed[0] == "l":
+            for motor in range(0, len(dict_motors[key_pressed[1]])-1):
+                self.mcuVAR.cmd_setMotorCalibrated(dict_motors[key_pressed[1]][motor],(2*i-1)*MULTIPLIER_PERCENT)
```

Comment:
What do you mean by i? Do you mean motor? Anyway, I'll let Lucas handle finding most of these small mistakes.